gpu/src/cuda/fft/ops.rs

```diff
@@ -51,23 +50,16 @@ pub fn gen_twiddles<F: IsFFTField>(
     function.retrieve_result()
 }
 
-// TODO: remove after implementing in cuda
-pub(crate) fn in_place_bit_reverse_permute<E>(input: &mut [E]) {
-    for i in 0..input.len() {
-        let bit_reversed_index = reverse_index(&i, input.len() as u64);
-        if bit_reversed_index > i {
-            input.swap(i, bit_reversed_index);
-        }
-    }
-}
+pub fn bitrev_permutation<F: IsFFTField>(
+    input: Vec<FieldElement<F>>,
+    state: &CudaState,
+) -> Result<Vec<FieldElement<F>>, CudaError> {
+    let result = input.clone();
```

Comment:
You could pass the same reference twice instead of cloning here. They are not mutable so it should be allowed.